sys/auto_init/auto_init.c

```diff
@@ -223,7 +223,7 @@ void auto_init(void)
     }
 
     /* initialize sensors and actuators */
-    if (IS_USED(MODULE_SHT1X)) {
+    if (IS_USED(MODULE_SHT1X) && IS_USED(MODULE_SHELL) && !IS_USED(MODULE_AUTO_INIT_SAUL)) {
```

Comment:
```suggestion
    if (IS_USED(MODULE_SHT1X)) {
```

This driver needs to be always auto-initialized, as other use if SAUL and direct use if the native driver interface are mutually exclusive, which would prevent using the shell app together with SAUL. See comment below.

Reply:
The change is intentional and reflect what is said in the comment: always build the auto_init function when the  used **and** the shell are used. It's skipped when saul is used as it's now handled in the saul auto_init code.

Reply:
The comment only explains why this driver has to be always auto initialized. The fact remains, this driver needs to be initialised no matter if shell and/or SAUL are used.

Reply:
The comment says `The sht1x module needs to be initialized regardless of SAUL being used, as the shell commands rely on auto-initialization`. So it's only needed when auto_init and the shell are used. Without saul, this driver only needs to be auto initialized when the shell is used. Not always. The change reflects that.